sklearn/model_selection/_search.py

```diff
@@ -520,6 +547,480 @@ def inverse_transform(self, Xt):
+
+    The parameters of the estimator used to apply these methods are optimized
+    by random search over parameter settings.
+
+    In contrast to UnsupervisedGridSearch, not all parameter values are tried out,
+    but rather a fixed number of parameter settings is sampled from the
+    specified distributions. The number of parameter settings that are tried
+    is given by n_iter.
+    
+    Unlike RandomizedSearchCV, no cross-validation is carried out. The parameters
+    found are the best for the input data and may not generalise to new examples.  
+
+    If all parameters are presented as a list,
+    sampling without replacement is performed. If at least one parameter
+    is given as a distribution, sampling with replacement is used.
+    It is highly recommended to use continuous distributions for continuous
+    parameters.
+
+    Read more in the :ref:`User Guide <randomized_parameter_search>`.
+
+    Parameters
+    ----------
+    estimator : estimator object.
+        A object of that type is instantiated for each grid point.
+        This is assumed to implement the scikit-learn estimator interface.
+        Either estimator needs to provide a ``score`` function,
+        or ``scoring`` must be passed.
+
+    param_distributions : dict
+        Dictionary with parameters names (string) as keys and distributions
+        or lists of parameters to try. Distributions must provide a ``rvs``
+        method for sampling (such as those from scipy.stats.distributions).
+        If a list is given, it is sampled uniformly.
+
+    n_iter : int, default=10
+        Number of parameter settings that are sampled. n_iter trades
+        off runtime vs quality of the solution.
+
+    scoring : string, callable or None, default=None
+        A string (see model evaluation documentation) or
+        a scorer callable object / function with signature
+        ``scorer(estimator, X)``.
+        If ``None``, the ``score`` method of the estimator is used.
+
+    fit_params : dict, optional
+        Parameters to pass to the fit method.
+
+    n_jobs : int, default=1
+        Number of jobs to run in parallel.
+
+    pre_dispatch : int, or string, optional
+        Controls the number of jobs that get dispatched during parallel
+        execution. Reducing this number can be useful to avoid an
+        explosion of memory consumption when more jobs get dispatched
+        than CPUs can process. This parameter can be:
+
+            - None, in which case all the jobs are immediately
+              created and spawned. Use this for lightweight and
+              fast-running jobs, to avoid delays due to on-demand
+              spawning of the jobs
+
+            - An int, giving the exact number of total jobs that are
+              spawned
+
+            - A string, giving an expression as a function of n_jobs,
+              as in '2*n_jobs'
+
+    refit : boolean, default=True
+        Refit the best estimator with the entire dataset.
```

Comment:
I don't think "entire" is appropriate here, because we're fitting with the entire dataset in all cases, are we not?

Indeed, it should be possible to retain (without refitting) the best model without any loss of efficiency, but that's not how things are currently implemented due to joblib not supporting an arbitrary reducer.